java/src/main/java/com/company/automation/fundamentals/optional_task/optional_task_01/OptionalTask01.java

```diff
@@ -0,0 +1,414 @@
+
+        String[] arrayByDecreasingLength;
+        arrayByDecreasingLength = sortByDecreasingLength(stringOfNumbers);
+        System.out.println("\nNumbers are sorted by decreasing their length:");
+        System.out.println(Arrays.toString(arrayByDecreasingLength));
+
+        printIntegersThatAreShorterThanAverageLength(stringOfNumbers);
+        printIntegersThatAreLongerThanAverageLength(stringOfNumbers);
+
+        System.out.println("\n" + Arrays.toString(stringOfNumbers));
+
+        int numberWithMinAmountOfDifferentDigits;
+        numberWithMinAmountOfDifferentDigits = getIntegerWithMinAmountOfDifferentDigits(stringOfNumbers);
+        System.out.println("\nThe first number that contains min amount of different digits: " +
+                numberWithMinAmountOfDifferentDigits + "\n");
+
+        int quantityOfNumbersWithOnlyEvenDigits;
+        quantityOfNumbersWithOnlyEvenDigits = getQuantityOfIntegersWithOnlyEvenDigits(stringOfNumbers);
+        System.out.println("Quantity of numbers which contain only even digits: " +
+                quantityOfNumbersWithOnlyEvenDigits + "\n");
+
+        int quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits;
+        quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits =
+                getQuantityOfIntegersWithEqualAmountOfEvenAndUnevenDigits(stringOfNumbers);
+        System.out.println("Quantity of numbers which contain equal amount of even and uneven digits: " +
+                quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits + "\n");
+
+        int integerWithRisingDigitsSequence;
+        integerWithRisingDigitsSequence = getIntegerWithRisingDigitsSequence(stringOfNumbers);
+        System.out.println("The first number with rising digits sequence: " + integerWithRisingDigitsSequence + "\n");
+
+        int integerWithOnlyDifferentDigits;
+        integerWithOnlyDifferentDigits = getIntegerWithOnlyDifferentDigits(stringOfNumbers);
+        System.out.println("The first number with only different digits: " + integerWithOnlyDifferentDigits + "\n");
+
+    }
+
+
+    private static String inputIntegersAsAString() {
+        boolean hasWrongCharacters;
+        boolean hasDigits;
+        String line;
+        String message;
+        Scanner in = new Scanner(System.in);
+
+        message = "Enter integer numbers using SPACE to divide them:>> ";
+        do {
+            System.out.print(message);
+            Pattern restrictedCharacters = Pattern.compile("[^0-9-\\s+]");
+            Pattern mustHave = Pattern.compile("[\\d]");
+            line = in.nextLine();
+            Matcher matcher = restrictedCharacters.matcher(line);
+            hasWrongCharacters = matcher.find();
+            matcher = mustHave.matcher(line);
+            hasDigits = matcher.find();
+        } while (hasWrongCharacters || !hasDigits);
+
+        line = line.replaceAll("\\+\\s*", " ");
+        line = line.replaceAll("-\\s*", " -");
+        line = line.trim();
+        return line;
```

Comment:
you can write this like 
return line.trim();

Reply:
I won't. It's not a big deal.
I suppose, If I do some manipulations with this String I should better  get this done before return.
I think it's more readable